hs_core/hydroshare/resource.py

```diff
@@ -577,57 +583,38 @@ def add_resource_files(pk, *files, **kwargs):
-    obsoleted, no other resources can obsolete it.
+    This does **not** handle mutability; changes to immutable resources should be denied elsewhere.
 
     """
     resource = utils.get_resource_by_shortkey(pk)
-    fed_res_file_names = kwargs.pop('fed_res_file_names', '')
     ret = []
-    fed_zone_home_path = kwargs.pop('fed_zone_home_path', '')
-    # for adding files to existing resources, the default action is copy
-    fed_copy_or_move = kwargs.pop('fed_copy_or_move', 'copy')
+    source_names = kwargs.pop('source_names', [])
+
+    if __debug__:
+        assert(isinstance(source_names, list))
+
+    move = kwargs.pop('move', False)
     folder = kwargs.pop('folder', None)
 
+    if __debug__:  # assure that there are no spurious kwargs left.
+        for k in kwargs:
+            print("kwargs[{}]".format(k))
+        assert len(kwargs) == 0
+
     for f in files:
-        if fed_zone_home_path:
-            # user has selected files from a federated iRODS zone, so files uploaded from local disk
-            # need to be stored to the federated iRODS zone rather than HydroShare zone as well
-            # TODO: why do we allow the federation prefix to vary in this statement?
-            ret.append(utils.add_file_to_resource(resource, f, folder=folder,
-                                                  fed_res_file_name_or_path=fed_zone_home_path))
-        elif resource.resource_federation_path:
-            # file needs to be added to a resource in a federated zone
-            # TODO: why do we allow the federation prefix to vary in this statement?
-            ret.append(utils.add_file_to_resource(
-                resource, f, folder=folder,
-                fed_res_file_name_or_path=resource.resource_federation_path))
-        else:
-            ret.append(utils.add_file_to_resource(resource, f, folder=folder))
-    if fed_res_file_names:
-        if isinstance(fed_res_file_names, basestring):
-            ifnames = string.split(fed_res_file_names, ',')
-        elif isinstance(fed_res_file_names, list):
-            ifnames = fed_res_file_names
-        else:
-            return ret
-        for ifname in ifnames:
-            ret.append(utils.add_file_to_resource(resource, None, folder=folder,
-                                                  fed_res_file_name_or_path=ifname,
-                                                  fed_copy_or_move=fed_copy_or_move))
+        ret.append(utils.add_file_to_resource(resource, f, folder=folder))
+
+    if len(source_names) > 0:
+        # TODO: eliminate string option; always pass a list. File names can contain ','!
+        for ifname in source_names:
+            ret.append(utils.add_file_to_resource(resource, None,
+                                                  folder=folder,
+                                                  source_name=ifname,
+                                                  move=move))
```

Comment:
@alvacouch The TO DO comment is a bit confusing here. Has the string option been eliminated? Looks like it has. If so, the TO DO comment needs to be updated. Also, a minor comment is that ```if len(source_names) > 0:``` can be changed to ```if source_names:``` to be simpler, but both will work.